sql: improve error message for ColumnConversionGeneral

otan opened the pull request:
Resolves https://github.com/cockroachdb/cockroach/issues/47141

Intended for backport to 20.1. Tagging Richard as a reviewer since he's working on SET DATA TYPE :P 

----

In addition to the release note, also modified TIME precision
conversions to use "General" which is the value for which means to
"overwrite" values.

Release note (sql change): Improved the error message for ALTER COLUMN
... SET DATA TYPE ... for data type conversions that involve overwriting
existing values.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/47155)
<!-- Reviewable:end -->

RichardJCai commented on pkg/sql/alter_table.go:
```diff
@@ -906,6 +906,16 @@ func applyColumnMutation(
+				9851,
+				fmt.Sprintf("%s->%s", col.Type.SQLString(), typ.SQLString()),
+				fmt.Sprintf(
+					"type conversion from %s to %s requires overwriting existing values which is not yet implemented",
```
I think you can use NewIssueWithDetailf instead of using fmt.Sprintf

RichardJCai reviewed (approved):
Looks good besides one small nit.

otan committed:
sql: improve error message for ColumnConversionGeneral

In addition to the release note, also modified TIME precision
conversions to use "General" which is the value for which means to
"overwrite" values.

Release note (sql change): Improved the error message for ALTER COLUMN
... SET DATA TYPE ... for data type conversions that involve overwriting
existing values.

otan commented:
bors r=RichardJCai
